### PR TITLE
Cypress Test for language selection and link redirection

### LINKED ISSUE #9017: Add New cypress Test for language selection and link redirection
**Description**

Add a new Cypress test to verify language selection and repository redirection functionality.

**Conditions to follow**

- Add these tests to the redirect.cy.ts file.
- Use the Page Object Model (POM) approach.
- Reuse existing functions and add necessary API verification with cy.intercept.

**Test - 1**

- After the LoginPage loads, click the "Contribute on GitHub" link and verify redirection by checking the URL and the new page content.
- Repeat the same for the "Third Party Software License" link.

**Test - 2**

- After the login page loads, switch between all five languages and verify that the login submit button text updates according to the selected language.
- Log in with any language selected other than English and verify that the sidebar items display in the chosen language

**Screenshot**

Image

### PR DESCRIPTION
**Pull request by Kaushikgtm**
## Proposed Changes

- Fixes #9017 
- Change 1 Added a Cypress test to verify that the login page's "Submit" button updates text dynamically when switching between different languages 
- Change 2 Created a test to log in with a non-English language (e.g., Hindi, Tamil, Kannada).
Verified that sidebar items display correctly in the chosen language.
- More?

@ohcnetwork/care-fe-code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA

## Summary by CodeRabbit

- **New Features**
	- Added language selection functionality with a dropdown on the login page.
	- Introduced new test cases for verifying localization and redirection links.
	- Enhanced element selection capabilities for sidebar and contribution links.
	- Added methods for verifying button texts and sidebar items based on language selection.

- **Bug Fixes**
	- Improved verification for sidebar items and button texts across multiple languages.

- **Documentation**
	- Updated interface for language-specific text mappings to improve clarity and usability.

### CODE REVIEW

**Comment by Jacobjeevan** [comment]
Check and implement the Coderabbit suggestions that are worth doing @Kaushikgtm

**Comment by nihal467** [comment]
@Kaushikgtm check the code rabbit suggestion and merge conflict issues

**Comment by Jacobjeevan** [comment]
Noticed that #9258 was a duplicate PR (closed). Please refer to the comments there and make changes accordingly 👍

**Comment by Kaushikgtm** [comment]
@Jacobjeevan I have made some changes please see

File: cypress/e2e/auth_spec/redirect.cy.ts [hunk 1]
```diff
@@ -1,10 +1,46 @@
 import LoginPage from "../../pageobject/Login/LoginPage";
```
Comment (Jacobjeevan) [inline cypress/e2e/auth_spec/redirect.cy.ts]: The imports are missing.

File: cypress/pageobject/Login/LoginPage.ts [hunk 2]
```diff
@@ -34,6 +37,62 @@ class LoginPage {
...
+
+  selectSidebarLanguage(languageCode: string) {
+    cy.get(this.languageSelector).select(languageCode);
+  }
+
+  verifySidebarText(expectedText: string) {
++    cy.get(this.sidebarSelector).should("have.text", expectedText);
+  }
+
+  switchLanguageAndVerifySidebars(languageMappings: { [key: string]: string }) {
```
Comment (Jacobjeevan) [inline cypress/pageobject/Login/LoginPage.ts]: The type here no longer matches what you are passing in.

File: cypress/pageobject/Login/LoginPage.ts [hunk 3]
```diff
@@ -1,6 +1,9 @@
 // LoginPage.ts
 
 class LoginPage {
+  submitButtonSelector = "#login";
```
Comment (Jacobjeevan) [inline cypress/pageobject/Login/LoginPage.ts]: Selectors are still missing.

**Review by Jacobjeevan — CHANGES_REQUESTED** [review]
Please make sure to test your changes before requesting for a review. Run cypress locally and ensure the tests are passing.

File: cypress/e2e/auth_spec/redirect.cy.ts [hunk 4]
```diff
@@ -1,10 +1,46 @@
...
 describe("redirect", () => {
+  const languageMappings = {
+  hi: hi["login"],
+  ta: ta["login"],
+  ml: ml["login"],
+  mr: mr["login"],
+  kn: kn["login"],
+  };
+  const languageSidebars = {
+    hi: {
```
Comment (Jacobjeevan) [inline cypress/e2e/auth_spec/redirect.cy.ts]: Don't need to keep repeating this either, since the structure is same for all languages.

**Review by Jacobjeevan — CHANGES_REQUESTED** [review]

**Comment by nihal467** [comment]
@Kaushikgtm Can you please share the current status of the PR? Otherwise, the PR will be closed in the next 48 hours, and you will be unassigned from the issue due to inactivity.

**Comment by Kaushikgtm** [comment]
@nihal467 I have update the PR, due to my semester exam i am inactive on this issue

**Comment by Kaushikgtm** [comment]
@Jacobjeevan, I have made some changes based on your suggestions

**Comment by Jacobjeevan** [comment]
> @Jacobjeevan, I have made some changes based on your suggestions

Some of the coderabbit suggestions are worth doing 👍  Implement those and resolve the merge conflicts.

**Comment by Kaushikgtm** [comment]
Hii @Jacobjeevan ,  i have made the changes in PR help me in resolve the merge conflicts

**Comment by Kaushikgtm** [comment]
@Jacobjeevan help me in resolved the merge conflicts

**Comment by Kaushikgtm** [comment]
Hii @Jacobjeevan Review my PR

**Comment by nihal467** [comment]
@Kaushikgtm can you clear the merge conflict

**Comment by Kaushikgtm** [comment]
@nihal467 I try but can't solve by my self please resolve merge conflict

**Comment by nihal467** [comment]
@Kaushikgtm post the message in care-general channel in slack, to seek help from where you are stuck

File: cypress/e2e/homepage_spec/redirect.cy.ts [hunk 5]
```diff
@@ -28,4 +49,28 @@ describe("redirect", () => {
     loginPage.ensureLoggedIn();
     cy.url().should("include", "/facility");
   });
+
+  it("Verify redirection of 'Contribute on GitHub' link", () => {
+    loginPage.clickContributeOnGitHub();
```
Comment (Jacobjeevan) [inline cypress/e2e/homepage_spec/redirect.cy.ts:54]: This won't work unless we disable chromeWebSecurity which is only applicable to Chrome anyways, so just verify the link.

```
cy.get(loginPage.contributeOnGitHubLink)
      .should("have.attr", "href", "https://github.com/ohcnetwork")
      .and("have.attr", "target", "_blank")
      .and("have.attr", "rel", "noopener noreferrer");
```

File: cypress/pageobject/Login/LoginPage.ts [hunk 6]
```diff
@@ -88,6 +111,68 @@ class LoginPage {
...
+  private verifySidebarElement(selector: string, expectedText: string): void {
+    cy.get(selector, { timeout: 10000 })
+      .should("be.visible")
+      .should("have.text", expectedText)
+      .should("not.be.disabled");
+  }
+
+  switchLanguageAndVerifySidebars(languageMappings: SidebarMapping): void {
+    Object.entries(languageMappings).forEach(([languageCode, texts]) => {
+      cy.get(this.languageSelector)
```
Comment (Jacobjeevan) [inline cypress/pageobject/Login/LoginPage.ts:162]: So language selector doesn't exist, from this code I assume you are referring to language selector from the profile page, but this test is not being called for a logged in user.

Additionally profile page was just replaced anyways. So you would need add logic for logging in and add the selector Id to the selection dropdown on profile page or rewrite the logic to use the language selection options on the login page.

File: cypress/pageobject/Login/LoginPage.ts [hunk 7]
```diff
@@ -88,6 +111,68 @@ class LoginPage {
...
+    return cy.get(this.submitButtonSelector);
+  }
+
+  getSidebarItems() {
+    return cy.get(`${this.sidebarSelectors.care}, ${this.sidebarSelectors.goal}, ${this.sidebarSelectors.footer_body}`);
+  }
+
+  switchLanguageAndVerifyButtonText(languageMappings: LanguageMapping): void {
+    Object.entries(languageMappings).forEach(([languageCode, texts]) => {
+      cy.get(this.languageSelector)
```
Comment (Jacobjeevan) [inline cypress/pageobject/Login/LoginPage.ts:137]: The logic below suggests you are trying to use language selector on profile page, but here it seems like you are referring to login page 😅

**Review by Jacobjeevan — CHANGES_REQUESTED** [review]

**Comment by Jacobjeevan** [comment]
Closing as per user's request.

**Pull request closed without merging**